integration-test.Rmd

```diff
@@ -190,43 +190,29 @@ Ensure this other configuration file also exists:
 
 ```{r}
 config_2 <- here("parameter_files", "WebParameters_2dii.yml")
-config_2_copy <- tempfile()
-fs::file_copy(config_2, config_2_copy)
 
 expect_true(file_exists(config_2))
 ```
 
 Ensure the paths in the configuration file work both locally and on GitHub actions:
 
 ```{r}
-make_config_portable <- function(config) {
-  lines <- readLines(config, encoding = "UTF-8")
-  lines <- make_paths_portable(lines)
-  writeLines(lines, config)
-
-  invisible(config)
-}
-
-make_paths_portable <- function(x) {
-  x %>%
-    root_field_path("project_location_ext", pattern = "PACTA_analysis") %>%
-    root_field_path("data_location_ext", pattern = "pacta-data") %>%
-    root_field_path("template_location", pattern = "create_interactive_report") %>%
-    root_field_path("stress_test_location", pattern = "StressTestingModelDev")
-}
-
-root_field_path <- function(x, field, pattern) {
-  parent <- path_dir(here())
-  value <- path(parent, extract_from(x, pattern))
-  sub(glue("({field}:[ ]?).*"), glue("\\1{value}/"), x)
+uses_relative_paths <- function(lines) {
+  patterns <- c(
+    "project_location_ext:[ ]?\\.\\./",
+    "data_location_ext:[ ]?\\.\\./",
+    "template_location:[ ]?\\.\\./",
+    "stress_test_location:[ ]?\\.\\./"
+  )
+  
+  all(purrr::map_lgl(patterns, ~any(grepl(.x, lines))))
 }
 
-extract_from <- function(x, pattern) {
-  line <- grep(pattern, x, value = TRUE)
-  sub(glue(".*({pattern}.*)"), "\\1", line)
+if (!uses_relative_paths(readLines(config_2))) {
+  abort("Pahts in the configuration file must use the form ../this_repo/")
 }
 
-make_config_portable(config_2)
+expect_true(uses_relative_paths(readLines(config_2)))
```

Comment:
Here I'm checking that the paths use the form "../this_repo" -- this is explained in a comment on the configuration file, but a test is better than a comment :smile: